src/backend/distributed/executor/citus_custom_scan.c

```diff
@@ -801,3 +817,53 @@ IsCitusCustomScan(Plan *plan)
 
 	return true;
 }
+
+
+/*
+ * In a Job, given a list of relations, if all them belong to the same
+ * colocation group, the Job's colocation ID is set to the group ID, else,
+ * it will be set to INVALID_COLOCATION_ID.
+ */
+static void
+SetJobColocationId(Job *job)
+{
+	uint32 jobColocationId = INVALID_COLOCATION_ID;
+
+	if (job->partitionKeyValue == NULL)
+	{
+		/* If the shard key is not known, colocationId is irrelevant */
+		job->colocationId = INVALID_COLOCATION_ID;
+		return;
+	}
+
+	List *rangeTableList = job->jobQuery->rtable;
+	ListCell *rangeTableCell = NULL;
+	foreach(rangeTableCell, rangeTableList)
```

Comment:
this seems insufficient, what if there are subqueries?

Reply:
Nope, good catch. We can subqueries by following sublinks, but now the bigger question, what else we might be missing? I need to think bit more.

Reply:
Changed to consider all the possible RTEs